designs/CentralNuGetManagement/CPVM-Project_Package_Version_Override.md

```diff
@@ -0,0 +1,425 @@
+
+`bar` and `baz` both have three dependencies. These are transitive dependencies for the project.
+
+- `foo` is imported with version 1.1.1 from the global `CentralManagementGroup.`
+    - `bar` and `baz` are in two different `CentralManagementGroup` that both have a `<PackageVersion>` for `foo`.
+    - Because both groups have `<PackageVersion>` for `foo`, the conflict is resolved by fallback to the global group which has a `<PackageVersion>` defined for `foo`.
+- `xyzzy` is imported with version 2.0.0 from `CentralManagementGroup="A"`.
+    - There is no `<PackageVersion>` defined for `xyzzy` in `CentralManagementGroup="B"` that `baz` belongs to.
+    - Because a `<PackageVersion>` is defined in only one group, that `CentralManagementGroup` can be resolved.
+    - Because `bar` is imported from `CentralManagementGroup="A"` where there is a `<PackageVersion>` for `xyzzy`, version 2.0.0 is resolved.
+- `qux` would throw a build error.
+    - A `<PackageVersion>` for `qux` is defined in both `CentralManagementGroup` `"A"` and `"B"`.
+    - No `<PackageVersion>` is defined for `qux` in the global group to resolve the conflict.
+    - `bar` has a direct dependency on `qux` version 202.0.0 and `baz` has a direct dependency on `qux` version 303.0.0. An error would be thrown due to a [cousin dependency](https://docs.microsoft.com/en-us/nuget/concepts/dependency-resolution#cousin-dependencies) conflict.
+
+### Sample Project 9: Undefined CentralManagementGroup Member
+
+9*.csproj*
+
+```xml
+<Project>
+  <ItemGroup>
+    <!-- No entry for Newtonsoft.Json in the "B" group -->
+    <!-- ❗ Throws an Error - Project Will Not Build ❗ -->
+    <PackageReference Include="Newtonsoft.Json" CentralManagementGroup="B" />
+  </ItemGroup>
+</Project>
+```
+
+Because there is no entry for `Newtonsoft.Json` in `CentralManagementGroup="B"`, the project will throw an error and fail to build.
+
+### Sample Project 10: Project Package Version Override Without Modifying Directory.Packages.props
+
+> Note:
+This is an anti-pattern, but it is described here for the benefit of developers who are unable to modify the Directory.Packages.props for their solution. Once the package version discrepancy is resolved or the Directory.Packages.props file can be updated, this implementation should be removed from your project.
+
+*10.csproj*
+
+```xml
+<Project>
+	<PropertyGroup>
+    <AllowProjectPackageVersion>
+        true
+    </AllowProjectPackageVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+		<!-- Remove version defined in Directory.Packages.props, if any --> 
+    <PackageVersion Remove="Newtonsoft.Json" /> 
+    <!-- Define PackageVersion which applies only to this project --> 
+    <PackageVersion Include="Newtonsoft.Json" Version="12.0.3" />
+    <!-- Version 12.0.3 -->
+    <PackageReference Include="Newtonsoft.Json" />
+  </ItemGroup>
+</Project>
+```
+
+Including a `<PackageVersion>` outside of the `Directory.Packages.props` file will ordinarily throw a build warning.
```

Comment:
I am not sure if MsBuild imports can be tracked. Same concern for the project data that we receive from ProjectSystem.